lib/kafka/consumer.rb

```diff
@@ -327,12 +335,12 @@ def make_final_offsets_commit!(attempts = 3)
       make_final_offsets_commit!(attempts - 1)
     end
 
-    def join_group
+    def join_group(clear_offsets_on_join: false)
       old_generation_id = @group.generation_id
 
       @group.join
 
-      if old_generation_id && @group.generation_id != old_generation_id + 1
+      if (old_generation_id && @group.generation_id != old_generation_id + 1) || clear_offsets_on_join
```

Comment:
i can moved this proposed change into a separate PR, that's why I kept it as the last commit. We would like to have the option of always clearing the offset but this change is not strictly required on this PR.